### ISSUE TITLE
Optional chaining et al

### ISSUE THREAD

**Issue by dy** [rollup/rollup#3469]
Browsers already support [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), but rollup doesn't recognize the syntax: 
```sh
[!] Error: Unexpected token
$.js (18:30)
16:         if (target.nodeType !== ELEMENT) return
17:         ;[target.classList.contains(SPECT_CLASS) ? target : null, ...target.getElementsByClassName(SPECT_CLASS)]
18:         .forEach(node => node?.[_spect].forEach(set => set.delete(node)))
                                  ^
19:       })
Error: Unexpected token
    at error (C:\projects\spect\node_modules\rollup\dist\shared\rollup.js:10120:30)
```

What's the right way to handle that? Hoped to avoid babel plugin.

**Comment by keithamus** [comment]
@dy you might want to follow https://github.com/acornjs/acorn/pull/891 which adds optional chaining to Acorn, the underlying parser rollup uses. Support for this will likely be blocked until Acorn gets it.

**Comment by memob0x** [comment]
i can reproduce this, rollup breaks when encounters obj?.prop syntax

**Comment by memob0x** [comment]
> Browsers already support [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), but rollup doesn't recognize the syntax:
> 
> ```shell
> [!] Error: Unexpected token
> $.js (18:30)
> 16:         if (target.nodeType !== ELEMENT) return
> 17:         ;[target.classList.contains(SPECT_CLASS) ? target : null, ...target.getElementsByClassName(SPECT_CLASS)]
> 18:         .forEach(node => node?.[_spect].forEach(set => set.delete(node)))
>                                   ^
> 19:       })
> Error: Unexpected token
>     at error (C:\projects\spect\node_modules\rollup\dist\shared\rollup.js:10120:30)
> ```
> 
> What's the right way to handle that? Hoped to avoid babel plugin.

i can reproduce it even with babel plugin tho

**Comment by lukastaegert** [comment]
This is available now in Rollup v2.11.0 via #3582
